ton_client/src/debot/routines.rs

```diff
@@ -130,3 +166,96 @@ pub(super) async fn sign_hash(ton: TonClient, arg: &str, signer: SigningBoxHandl
     .map_err(|err| format!("Can not sign hash: {}", err))?;
     Ok(result.signature)
 }
+
+pub(super) fn generate_random(ton: TonClient, args: serde_json::Value) -> Result<String, String> {
+    let len_str = args["length"]
+        .as_str()
+        .ok_or(format!(r#""len" not found"#))?;
+    let len =
+        u32::from_str_radix(len_str, 10).map_err(|e| format!("failed to parse length: {}", e))?;
+    let result = generate_random_bytes(ton, ParamsOfGenerateRandomBytes { length: len })
+        .map_err(|e| format!(" failed to generate random: {}", e))?;
+    Ok(result.bytes)
+}
+
+fn get_arg(args: &serde_json::Value, name: &str) -> Result<String, String> {
+    args[name]
+        .as_str()
+        .ok_or(format!("\"{}\" not found", name))
+        .map(|v| v.to_string())
+}
+
+pub(super) fn nacl_box(ton: TonClient, args: serde_json::Value) -> Result<String, String> {
+    let public = decode_abi_bigint(&get_arg(&args, "publicKey")?).map_err(|e| e.to_string())?;
+    let secret = decode_abi_bigint(&get_arg(&args, "secretKey")?).map_err(|e| e.to_string())?;
+    let result = crate::crypto::nacl_box(
+        ton,
+        ParamsOfNaclBox {
+            decrypted: base64::encode(&get_arg(&args, "decrypted")?),
+            nonce: get_arg(&args, "nonce")?,
+            their_public: hex::encode(public.to_bytes_be().1),
+            secret: hex::encode(secret.to_bytes_be().1),
+        },
+    )
+    .map_err(|e| format!(" failed to encrypt with nacl box: {}", e))?;
+    Ok(result.encrypted)
+}
+
+pub(super) fn nacl_box_gen_keypair(
+    ton: TonClient,
+    args: serde_json::Value,
+) -> Result<KeyPair, String> {
+    let secret = decode_abi_bigint(&get_arg(&args, "secret")?).map_err(|e| e.to_string())?;
+    let result = nacl_box_keypair_from_secret_key(
+        ton,
+        ParamsOfNaclBoxKeyPairFromSecret {
+            secret: hex::encode(secret.to_bytes_be().1),
+        },
+    )
+    .map_err(|e| format!(" failed to generate keypair from secret: {}", e))?;
+    Ok(result)
+}
+
+pub(super) async fn get_account_state(
+    ton: TonClient,
+    args: serde_json::Value,
+) -> Result<serde_json::Value, String> {
+    let addr = get_arg(&args, "addr")?.to_lowercase();
+    let mut accounts = query_collection(
+        ton,
+        ParamsOfQueryCollection {
+            collection: "accounts".to_owned(),
+            filter: Some(json!({
```

Comment:
Get only `boc` field here and then parse it with `boc::parse_account` function

Reply:
Done.